### PR TITLE
src/cyw43_ll.c: Fix some build warnings.

### LINKED ISSUE #115: Code builds with warnings v1.0.3
I'm trying to update the cyw43-driver in pico-sdk. We have a test that everything builds without warnings and errors. There are some new issues in the latest version of the cyw43_driver v1.03.

```
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c: In function 'cyw43_check_valid_chipset_firmware':
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:367:60: error: unused parameter 'self' [-Werror=unused-parameter]
  367 | static int cyw43_check_valid_chipset_firmware(cyw43_int_t *self, size_t len, uintptr_t source) {
      |                                               ~~~~~~~~~~~~~^~~~
In file included from /home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:40:
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c: In function 'cyw43_ll_sdpcm_poll_device':
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:25: error: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~
      |                                                           |
      |                                                           uint32_t {aka long unsigned int}
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_config.h:122:34: note: in definition of macro 'CYW43_PRINTF'
  122 | #define CYW43_PRINTF(...) printf(__VA_ARGS__)
      |                                  ^~~~~~~~~~~
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:13: note: in expansion of macro 'CYW43_DEBUG'
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |             ^~~~~~~~~~~
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:53: note: format string is defined here
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |                                                    ~^
      |                                                     |
      |                                                     unsigned int
      |                                                    %lu
cc1: all warnings being treated as errors
```

### PR DESCRIPTION
**Pull request by peterharperuk**
Fixes #115

### CODE REVIEW

**Comment by peterharperuk** [comment]
@dpgeorge Any chance of getting this merged?

**Comment by peterharperuk** [comment]
Sorry to nag, but I can't update the cyw43-firmware in the pico-sdk with build warnings, so need these fixed.

File: src/cyw43_ll.c [hunk 1]
```diff
@@ -364,7 +364,7 @@ static void cyw43_write_backplane(cyw43_int_t *self, uint32_t addr, size_t size,
     cyw43_set_backplane_window(self, CHIPCOMMON_BASE_ADDRESS);
 }
 
-static int cyw43_check_valid_chipset_firmware(cyw43_int_t *self, size_t len, uintptr_t source) {
+static int cyw43_check_valid_chipset_firmware(size_t len, uintptr_t source) {
```
Comment (dpgeorge) [inline src/cyw43_ll.c:367]: I would prefer to add a line at the start of this function like `(void)self;`.  That's how this is handled elsewhere in this driver.

**Comment by dpgeorge** [comment]
The CI for this repo does build the code with `-Wpedantic` enabled, but that didn't pick up the warnings.  Is there any way to make the CI hit at least the unused variable warning?

File: src/cyw43_ll.c (hunk 1 above)
Comment (peterharperuk) [reply src/cyw43_ll.c]: done

**Comment by dpgeorge** [comment]
Thanks for updating, now merged.

**Pull request merged**